docs/DeveloperGuide.md

```diff
@@ -1,34 +1,972 @@
-# Developer Guide
+# iGraduate Developer Guide
+By: `W09-2` Latest update: `30 March 2021`
 
-## Design & implementation
+- [iGraduate Developer Guide](#igraduate-developer-guide)
+    * [1. Introduction](#1-introduction)
+    * [2. Setting up, getting started](#2-setting-up-getting-started)
+        + [2.1 Terminal](#21-terminal)
+        + [2.2 Intellij IDEA (Recommended)](#22-intellij-idea-recommended)
+    * [3. Design](#3-design)
+        + [3.1 Architecture](#31-architecture)
+        + [3.2 UI Component](#32-ui-component)
+        + [3.3 Logic Component](#33-logic-component)
+        + [3.4 Model Component](#34-model-component)
+            + [3.4.1 `module` Package](#341-module-package)
+                + [3.4.1.1 `Module` Class](#3411-module-class)
+                + [3.4.1.2 `CoreModule` Class](#3412-coremodule-class)
+                + [3.4.1.3 `GeModule` Class](#3413-gemodule-class)
+                + [3.4.1.4 `ElectiveModule` Class](#3414-electivemodule-class)
+                + [3.4.1.5 `MathModule` Class](#3415-mathmodule-class)
+            + [3.4.2 `list` Package](#342-list-package)
+                + [3.4.2.1 `ModuleList` Class](#3421-modulelist-class)
+        + [3.5 Storage Component](#35-storage-component)
+        + [3.6 Common Classes](#36-common-classes)
+    * [4. Implementation](#4-implementation)
+        + [4.1 UI](#41-ui)
+        + [4.2 Parser](#42-parser)
+        + [4.3 Command](#43-command)
+        + [4.4 Module](#44-module)
+        + [4.5 ModuleList](#45-modulelist)
+        + [4.6 Storage](#46-storage)
+        + [4.7 Exception](#47-exception)
+    * [Appendix: Requirements](#appendix-requirements)
+        + [Product Scope](#product-scope)
+            + [Target User Profile](#target-user-profile)
+            + [Value Proposition](#value-proposition)
+        + [User Stories](#user-stories)
+    * [Instructions for Manual Testing](#instructions-for-manual-testing)
+        + [Launch and Shutdown](#launch-and-shutdown)
+        + [Adding a module](#adding-a-module)
+        + [Deleting a module](#deleting-a-module)
+        + [Marking modules as done](#marking-modules-as-done)
+        + [Updating the module list](#updating-the-module-list)
+        + [CAP](#cap)
+        + [Progress](#progress)
+        + [List Modules](#list-modules)
+        + [Saving data](#saving-data)
+## 1. Introduction
```

Comment:
Did you intent for introduction to be a main header instead? 
![image](https://user-images.githubusercontent.com/58136829/113231561-18af6c00-92ce-11eb-808b-1b6b484aabf9.png)